### DISCUSSION TITLE
Async perfomance

### DISCUSSION THREAD

**Discussion by Valorix** [php-rdkafka/php-rdkafka#519] [Q&A]
Hi,

I have developed a script that produce a simple message to kafka, 
I'm surprised about the time to execute, around 1 sec.
Is the function are really async ? Why I didn't have an immediate return ?

Thanks

**Comment by nick-zh** [comment]
Initial broker connection might take some time, as long as you use `poll(0)` it is non-blocking and if you are not flushing after each message
